src/pages/home/report/ContextMenu/ContextMenuActions.tsx

```diff
@@ -542,7 +543,21 @@ const ContextMenuActions: ContextMenuAction[] = [
         icon: Expensicons.Copy,
         successTextTranslateKey: 'reportActionContextMenu.copied',
         successIcon: Expensicons.Checkmark,
-        shouldShow: (type) => type === CONST.CONTEXT_MENU_TYPES.REPORT && (Environment.isDevelopment() || Environment.isStaging() || Environment.isInternalTestBuild()),
+        shouldShow: (type) => {
+            // eslint-disable-next-line react-hooks/rules-of-hooks
+            const {environment, isProduction} = useEnvironment();
```

Comment:
The hook should land on the top level of the component

Reply:
There is no component here since ContextMenuActions is just an array of objects, that's why I had to go along with the eslint disable

Reply:
I've moved it out to the parent component `BaseReportActionContextMenu` since `ContextMenuActions` itself is just a list